src/stubgen.py

```diff
@@ -279,7 +331,7 @@ def put_nb_func(self, fn, name: Optional[str] = None) -> None:
                 self.write_ln(f"@{overload}")
                 self.put_nb_overload(fn, s, name)
 
-    def put_function(self, fn, name: Optional[str] = None, parent=None):
+    def put_function(self, fn: NbFunction, name: Optional[str] = None, parent: Optional[object] = None):
```

Comment:
This function is also used to process pure Python functions without any of the nanobind bits. I am surprised if this type-checks with the `NbFunction` signature.

Reply:
See also this.